animation-proposal.md

```diff
@@ -0,0 +1,338 @@
+	 * Equivalent to: [0.42, 0.0, 0.58, 1.0]
+	 */
+	| "ease-in-out"
+
+	/**
+	 * The animation starts abruptly, and then progressively slows down
+	 * towards the end.
+	 * Equivalent to: [0.0, 0.0, 0.58, 1.0]
+	 */
+	| "ease-out"
+
+	/**
+	 * Define a custom curve by providing P1 and P2 of a cubic bezier
+	 * curve. To be valid, each point's abscissa must be in [0, 1].
+	 * The ordinate may be outside this range; if so then you will
+	 * get a bounce effect.
+	 */
+	| number[];
+
+/**
+ * Describes how the `value` parameter to `interpolate` should be interpreted.
+ */
+export type InterpolationBasis
+	/** The value is an absolute value. The function operates as "Interpolate to this value". */
+	= "absolute"
+
+	/** The value is relative to the starting value. The function operates as "Interpolate by this much". */
+	| "relative"
+	;
+
+/**
+ * Describes how many times the interpolation should repeat.
+ */
+export type InterpolationIterationCount
+	/** Run the interpolation this many times. */
+	= number
+
+	/** Run the interpolation forever (until canceled). */
+	| "infinite"
+	;
+
+/**
+ * Specifies the direction the interpolation should be played.
+ */
+export type InterpolationDirection
+	/** The interpolation is played forward. */
+	= "forward"
+	/** The interpolation is played in reverse. */
+	| "reverse"
+	/** The interpolation is played forward, then backward, alternating. */
+	| "alternate"
+	/** Same as "alternate", but starts out playing backward. */
+	| "alternate-reverse"
+	;
+```
+
+##### Examples
+
+```ts
+// Interpolate a vector (lerp).
+this.context.interpolate(actor.selector("transform.app.position"), position, 0.2);
+
+// Interpolate a quaternion (slerp).
+this.context.interpolate(actor.selector("transform.app.rotation"), rotation, 0.2, { curve: "ease" });
+
+// Interpolate a number.
+this.context.interpolate(actor.selector("light.intensity"), 1, 0.2, { curve: "ease-in" });
+
+// Interpolate a color.
+this.context.interpolate(material.selector("color"), color, 0.2, { curve: "ease-out" });
+
+// Interpolate a value on a custom curve (a "bounce", in this case).
+this.context.interpolate(material.selector("color"), color, 0.2, { curve: [0.1, -0.6, 0.2, 0] });
+
+// Interpolate to another actor's position by reference. The host reads the
+// target actor's position locally via the target selector, enabling smooth
+// continuous interpolation to a moving target, and alleviating the need to
+// subscribe to the target actor's transform.
+this.context.interpolate(
+	actor.selector("transform.app.position"),
+	target.selector("transform.app.position"),
+	0.2, {
+		iterationCount: "infinite"
+	});
+
+// Cancel an interpolation. This applies to all types of interpolations, but
+// is particularly useful for stopping continuous interpolations.
+this.context.interpolate(
+	actor.selector("transform.app.position"),
+	target.selector("transform.app.position"),
+	null);
+)
+```
+
+#### Add `Actor.selector` and `Asset.selector` methods
```

Comment:
This is basically the same function on both classes, right? Could it be a utility method instead of putting it in the prototype?

Reply:
They do the same thing, but the root of the selector string they generate is different, and they may need to access private info. A selector generated by an Actor will look something like: "actor.<actorId>.transform.app.position", whereas a Material might generate:
"material.<materialId>.<containerId>.mainTextureOffset.u".